### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by abhishektanwar**
FUNCTIONALITIES ADDED :
- Login (data persists on reload)
- Register
- Logout

FILES TO CHECK :
 - src/components/PrivateRoute/PrivateRoute.jsx
 - src/components/Authentication/SignIn.jsx
 - src/components/Authentication/SignUp.jsx
 - src/reducers/userSlice.js
 - src/store/store.js
 - src/App.js

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -4,14 +4,33 @@ import Login from './pages/login/Login'
...
+import PrivateRoute from './components/PrivateRoute/PrivateRoute';
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { verifyUser } from './reducers/userSlice';
+import AuthRoute from './components/AuthRoute/AuthRoute';
+import utils from './utils';
 function App() {
+  const {user,authToken} = useSelector((state)=>state.user);
+  const dispatch = useDispatch();
+// trekgram-auth-token
```
Comment (iRohitGaur) [inline src/App.js]: suggestion: remove comments

File: src/components/AuthRoute/AuthRoute.jsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { useSelector } from "react-redux";
+import { Navigate, Outlet, useLocation } from "react-router-dom";
+import utils from "../../utils";
+// import utils from "../../utils";
+
+const AuthRoute = () => {
+  // const trekToken = utils.getLocalStorage('trek-notes-authToken')
```
Comment (iRohitGaur) [inline src/components/AuthRoute/AuthRoute.jsx]: suggestion: remove commented code

File: src/components/Authentication/SignIn.jsx [hunk 3]
```diff
@@ -30,10 +34,20 @@ const SignIn = () => {
...
+  const loginUser = () => {
+    dispatch(login(loginCredentials))
+    
+  };
+
+  // useEffect(()=>{
+  //   if(user && authToken){
+  //     navigate("/");
+  //   }
+  // },[user])
```
Comment (iRohitGaur) [inline src/components/Authentication/SignIn.jsx]: suggestion: remove commented code

File: src/components/PrivateRoute/PrivateRoute.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { useSelector } from "react-redux";
+import { Navigate, Outlet, useLocation } from "react-router-dom";
+import utils from "../../utils";
+// import utils from "../../utils";
+
+const PrivateRoute = () => {
+  // const trekToken = utils.getLocalStorage('trek-notes-authToken')
```
Comment (iRohitGaur) [inline src/components/PrivateRoute/PrivateRoute.jsx]: suggestion: remove commented code

**Review by iRohitGaur — APPROVED** [review]
Nice work!

Gave some suggestions, please look into it.

Good to merge with updates 👍🏻

File: .env [hunk 5]
```diff
@@ -0,0 +1,2 @@
+LOCAL_STORAGE_KEY=TREKGRAM_AUTH_TOKEN
+BASE_API_URL=https://trekgram-backend.herokuapp.com
```
Comment (bhushannaikaroskar) [inline .env:2]: I think you should exclude .env file as it contains secret key. Add this .env file to gitignore

**Review by bhushannaikaroskar — APPROVED** [review]
Hey abhishek reviewed your code. Code is really clean and understandable. Please check suggestion before merging. Good to merge

File: src/components/PrivateRoute/PrivateRoute.jsx (hunk 4 above)
Comment (abhishektanwar) [reply src/components/PrivateRoute/PrivateRoute.jsx]: removed

File: src/components/Authentication/SignIn.jsx (hunk 3 above)
Comment (abhishektanwar) [reply src/components/Authentication/SignIn.jsx]: removed

File: src/components/AuthRoute/AuthRoute.jsx (hunk 2 above)
Comment (abhishektanwar) [reply src/components/AuthRoute/AuthRoute.jsx]: removed.

File: src/App.js (hunk 1 above)
Comment (abhishektanwar) [reply src/App.js]: removed

**Pull request merged**